py/h2o_wave/h2o_wave/cli.py

```diff
@@ -276,4 +284,70 @@ def learn():
         from h2o_wave_university import cli
         cli.main()
     except ImportError:
-        print('You need to run \x1b[7;30;43mpip install h2o_wave_university\x1b[0m first.')
\ No newline at end of file
+        print('You need to run \x1b[7;30;43mpip install h2o_wave_university\x1b[0m first.')
+
+
+@main.command()
+@click.option('--port', default=10101, help='Port your app is running on (defaults to 10101).')
+@click.option('--subdomain', default='?new', help='Subdomain to use. If not available, a random one is generated.')
+@click.option('--remote-host', default='h2oai.app', help='Remote host to use (defaults to h2oai.app).')
+def share(port: int, subdomain: str, remote_host: str):
+    """Share your locally running app with the world.
+
+    \b
+    $ wave share
+    """
+    if 'Windows' in platform.system():
+        asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+
+    if 'Windows' in platform.system():
+
+        async def wakeup():
+            while True:
+                await asyncio.sleep(1)
+
+        # HACK: Enable Ctrl-C on Windows when opening multiple TCP connections.
+        # https://stackoverflow.com/questions/27480967/why-does-the-asyncios-event-loop-suppress-the-keyboardinterrupt-on-windows.
+        loop.create_task(wakeup())
+
+    try:
+        loop.run_until_complete(_share(port, subdomain, remote_host))
+    except KeyboardInterrupt:
+        tasks = asyncio.all_tasks(loop)
+        for task in tasks:
+            task.cancel()
+        loop.run_until_complete(asyncio.gather(*tasks, return_exceptions=True))
+        loop.close()
+
+
+async def _share(port: int, subdomain: str, remote_host: str):
+    if _scan_free_port(port) == port:
+        print(f'Could not connect to localhost:{port}. Please make sure your app is running.')
+        exit(1)
+
+    res = httpx.get(f'https://{remote_host}/{subdomain}', headers={'Content-Type': 'application/json'})
+    if res.status_code != 200:
+        print('Could not connect to the remote sharing server.')
+        exit(1)
+
+    res = res.json()
+    print(f'BETA: Proxying localhost:{port} ==> {res["url"]}')
+    print('\x1b[7;30;43mDO NOT SHARE YOUR APP IF IT CONTAINS SENSITIVE INFO\x1b[0m.')
+    print('Press Ctrl+C to stop sharing.')
+
+    max_conn_count = res['max_conn_count']
+    step = 100 if max_conn_count > 10 else max_conn_count
```

Comment:
Is this condition alright? Looks fishy.

Reply:
Good question. This piece probably deserves a comment in the code with further explanations. It's there because there are 2 possible server configurations:

1. The "community" one used for regular Wave users - limits to 10 concurrent connections per client.
2. Using this reverse proxy to tunnel a single app only with significantly more concurrent connections (10k+).

The condition is there simply to create the server connections faster (step of 100 instead of 10 => effectively 10x faster).

Does it make sense?